src/components/organisms/RolesTabs/RolesTabs.tsx

```diff
@@ -108,6 +114,7 @@ const RolesTabs: FC = () => {
             {...role}
             temporaryName={tabNames[role.id]}
             allPrivileges={allPrivileges}
+            isMainUser={isMainUser}
```

Comment:
**[blocker]**

Currently it seems that what you are checking whether the role has "is_root" = true or not. Current approach is 

1. Pick all roles that have is_root="true"
2. Create an array of all their id's
3. While mapping over all roles, check if their id is in that list

This seems to be a bit overengineered.
We are already mapping over all roles, when rendering the list of <RoleForm> components.
So the part we spread there (...role) also includes the "is_root" value for that form. I would suggest using the value from there directly, instead of passing a duplicate of the same value. (google the KISS principle) - Essentially do things in as simple a way as possible.

Reply:
Good point yes, changed it